backend/substrapp/tasks/tasks.py

```diff
@@ -217,7 +214,7 @@ def fetch_model(channel_name, parent_tuple_type, authorized_types, input_model,
         get_and_put_model_content(
             channel_name,
             tuple_type,
-            input_model['traintuple_key'] + HASH_KEY_SUFFIX_TRUNK,
+            input_model['traintuple_key'],
```

Comment:
Can you explain why we can safely remove the hash suffix here ?

Reply:
It is not needed anymore because primary keys are not based on model contents anymore. As a result, there's no possibility of a primary key conflict, hence no need for a hash salt.

This feature (hash suffixes) was implemented to support the edge case when the trunk and head models are identical.

More generally, it's now ok to save duplicate models.